test/e2e/util.go

```diff
@@ -560,77 +577,84 @@ func RunRC(c *client.Client, name string, ns, image string, replicas int) error
 	last = 0
 	failCount = 10
 	current = 0
-	oldPods := &api.PodList{}
+	var oldPods []api.Pod
+	podLists.Reset()
+	foundAllPods := false
 	for same < failCount && current < replicas {
-		current = 0
-		waiting := 0
-		pending := 0
-		unknown := 0
-		inactive := 0
-		time.Sleep(10 * time.Second)
-
-		// TODO: Use a reflector both to put less strain on the cluster and
-		// for more clarity.
-		currentPods, err := listPods(c, ns, label, fields.Everything())
-		if err != nil {
-			return fmt.Errorf("Error listing pods: %v", err)
-		}
-		for _, p := range currentPods.Items {
-			if p.Status.Phase == api.PodRunning {
-				current++
-				if err := VerifyContainersAreNotFailed(p, maxContainerFailures); err != nil {
-					return err
+		time.Sleep(2 * time.Second)
+		for podLists.Len() > 0 && current < replicas {
```

Comment:
Can you please add a comment that you read all available elements without sleeping? It was not obvious for me.